web-app/src/lib/job-input/job-input.ts

```diff
@@ -0,0 +1,84 @@
+import { z } from "zod";
+
+import { JobInputSchemaIntlPath, ValidJobInputTypes } from "./type";
+import { isTypeAndValidationValid } from "./util";
+import { validationSchema } from "./validation";
+
+export const jobInputSchema = (t?: IntlTranslation<JobInputSchemaIntlPath>) =>
+  z
+    .object({
+      id: z.string().nonempty({
+        message: t?.("Id.required"),
+      }),
+      type: z.enum(ValidJobInputTypes, {
+        message: t?.("Type.enum", {
+          options: ValidJobInputTypes.join(", "),
+        }),
+      }),
+      name: z
+        .string()
+        .nonempty({
+          message: t?.("Name.required"),
+        })
+        .min(2, { message: t?.("Name.min") })
+        .max(128, {
+          message: t?.("Name.max"),
+        }),
+      data: z.object({
+        values: z
+          .array(
+            z.string().min(1, {
+              message: t?.("Data.Values.value.required"),
+            }),
+          )
+          .min(1, { message: t?.("Data.Values.min") })
+          .optional(),
+        placeholder: z
+          .string()
+          .nonempty({
+            message: t?.("Data.Placeholder.required"),
+          })
+          .min(2, { message: t?.("Data.Placeholder.min") })
+          .max(128, {
+            message: t?.("Data.Placeholder.max"),
+          }),
```

Comment:
The placeholder is optional and should be checked for `string` and `nonempty`.

Reply:
@golddydev It should be fixed with 85af789f801e667d5c553a50c3045bc18290455e.

Please double-check and feel free to resolve the discussion if you agree with the change.